_posts/2022-04-21-ambient-peer-discovery.md

```diff
@@ -0,0 +1,334 @@
+---
+layout: post
+name:  "Waku v2 Ambient Peer Discovery"
+title:  "Waku v2 Ambient Peer Discovery"
+date: 2022-04-21 10:00:00 +0200
+author: Daniel
+published: true
+permalink: /wakuv2-apd-2022-21
+categories: research
+summary: Introducing and discussing ambient peer discovery methods currently used by Waku v2, as well as future plans regarding ambient peer discovery.
+image: /assets/img/waku_v2_discv5_random_walk_estimation.svg
+discuss:
+---
+
+[Waku v2](https://rfc.vac.dev/spec/10/) comprises a set of modular protocols for secure privacy preserving communication.
+Avoiding centralization, these protocols exchange messages over a P2P network layer.
+In order to build a P2P network, participating nodes first have to discover peers within this network.
+This is where [*ambient peer discovery*](https://docs.libp2p.io/concepts/publish-subscribe/#discovery) comes into play:
+it allows nodes to find peers, making it an integral part of any decentralized application.
+
+In this post the term *node* to refers to *our* endpoint or the endpoint that takes action,
+while the term *peer* refers to other endpoints in the P2P network.
+These endpoints can be any device connected to the Internet: e.g. servers, PCs, notebooks, mobile devices, or applications like a browser.
+As such, nodes and peers are the same. We use these terms for the ease of explanation without loss of generality.
+
+
+In Waku's modular design, ambient peer discovery is an umbrella term for mechanisms that allow nodes to find peers.
+Various ambient peer discovery mechanisms are supported, and each is specified as a separate protocol.
+Where do these protocols fit into Waku's protocol stack?
+The P2P layer of Waku v2 builds on [libp2p gossipsub](https://github.com/libp2p/specs/blob/10712c55ab309086a52eec7d25f294df4fa96528/pubsub/gossipsub/README.md).
+Nodes participating in a gossipsub protocol manage a mesh network that is used for routing messages.
```

Comment:
Would "form" perhaps be a more accurate word choice than "manage" here?

Reply:
Agreed, it read more fluidly.
I used manage, because the nodes do manage various sets of peers: `mesh`, `fanout`, `peers`.
This illustrates that mesh management and peer discovery are in different layers. Wdyt?